lib/src/analyzers/unused_l10n_analyzer/unused_l10n_analyzer.dart

```diff
@@ -128,40 +129,71 @@ class UnusedL10nAnalyzer {
     final unusedLocalizationIssues = <UnusedL10nFileReport>[];
 
     localizationUsages.forEach((classElement, usages) {
-      final unit = classElement.thisOrAncestorOfType<CompilationUnitElement>();
-      if (unit != null) {
-        final lineInfo = unit.lineInfo;
-        if (lineInfo != null) {
-          final accessorSourceSpans =
-              _getUnusedAccessors(classElement, usages, unit);
-          final methodSourceSpans =
-              _getUnusedMethods(classElement, usages, unit);
-
-          final filePath = unit.source.toString();
-          final relativePath = relative(filePath, from: rootFolder);
-
-          final issues = [
-            ...accessorSourceSpans,
-            ...methodSourceSpans,
-          ];
-
-          if (issues.isNotEmpty) {
-            unusedLocalizationIssues.add(
-              UnusedL10nFileReport(
-                path: filePath,
-                relativePath: relativePath,
-                className: classElement.name,
-                issues: issues,
-              ),
-            );
-          }
+      _getUnusedReports(
+        classElement,
+        unusedLocalizationIssues,
+        usages,
+        rootFolder,
+      );
+
+      final hasCustomSupertype = classElement.allSupertypes.length > 1;
+      if (hasCustomSupertype) {
+        final supertype = classElement.supertype;
+        if (supertype is InterfaceType) {
+          _getUnusedReports(
+            supertype.element,
+            unusedLocalizationIssues,
+            usages,
+            rootFolder,
+            overriddenClassName: classElement.name,
+          );
         }
       }
     });
 
     return unusedLocalizationIssues;
   }
 
+  void _getUnusedReports(
+    ClassElement classElement,
+    List<UnusedL10nFileReport> reports,
+    Iterable<String> usages,
+    String rootFolder, {
+    String? overriddenClassName,
+  }) {
+    final unit = classElement.thisOrAncestorOfType<CompilationUnitElement>();
+    if (unit == null) {
+      return;
+    }
+
+    final lineInfo = unit.lineInfo;
+    if (lineInfo == null) {
+      return;
+    }
+
+    final accessorSourceSpans = _getUnusedAccessors(classElement, usages, unit);
+    final methodSourceSpans = _getUnusedMethods(classElement, usages, unit);
+
+    final filePath = unit.source.toString();
+    final relativePath = relative(filePath, from: rootFolder);
+
+    final issues = [
+      ...accessorSourceSpans,
+      ...methodSourceSpans,
+    ];
+
+    if (issues.isNotEmpty) {
+      reports.add(
```

Comment:
@incendial can we return object instead add them to passed list? This looks like side effect.

Reply:
Fixed